File: app/src/test/java/com/annushkaproject/programmerscalculator/ProgrammerTests/ProgrammerModTest.java
package com.annushkaproject.programmerscalculator.ProgrammerTests;

import com.annushkaproject.programmerscalculator.model.ProgrammerCalcModel;
import com.annushkaproject.programmerscalculator.model.Value;
import com.annushkaproject.programmerscalculator.model.WordLength;
import com.annushkaproject.programmerscalculator.utils.ProgrammerOperationsUtil;

import org.junit.Test;

import static com.annushkaproject.programmerscalculator.model.Operator.MOD;
import static org.junit.Assert.*;

public class ProgrammerModTest {
    @Test
    public void modTestAll_isCorrect() {
        for (WordLength mode: WordLength.values()) {
            assertEquals(1, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new Value(3), new Value(2), MOD, mode)));
            assertEquals(0, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new Value(2), new Value(2), MOD, mode)));
            assertEquals(0, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new Value(0), new Value(2), MOD, mode)));
            assertEquals(-1, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new Value(-3), new Value(2), MOD, mode)));
            assertEquals(-1, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new Value(-3), new Value(-2), MOD, mode)));
            assertEquals(0, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new Value(-2), new Value(-2), MOD, mode)));
        }
    }
}

File: app/src/test/java/com/annushkaproject/programmerscalculator/ProgrammerTests/ProgrammerModTest.kt
package com.annushkaproject.programmerscalculator.ProgrammerTests

import com.annushkaproject.programmerscalculator.model.Operator
import com.annushkaproject.programmerscalculator.model.ProgrammerCalcModel
import com.annushkaproject.programmerscalculator.model.int_size_enum
import com.annushkaproject.programmerscalculator.utils.ProgrammerOperationsUtil.calculateWithData
import org.junit.Assert
import org.junit.Test
import java.math.BigDecimal

class ProgrammerModTest {
    @Test
    fun modTestAll_isCorrect() {
        for (mode in int_size_enum.values()) {
            Assert.assertEquals(1, calculateWithData(ProgrammerCalcModel(BigDecimal(3), BigDecimal(2), Operator.get_remainder, mode)))
            Assert.assertEquals(0, calculateWithData(ProgrammerCalcModel(BigDecimal(2), BigDecimal(2), Operator.get_remainder, mode)))
            Assert.assertEquals(0, calculateWithData(ProgrammerCalcModel(BigDecimal(0), BigDecimal(2), Operator.get_remainder, mode)))
            Assert.assertEquals(-1, calculateWithData(ProgrammerCalcModel(BigDecimal(-3), BigDecimal(2), Operator.get_remainder, mode)))
            Assert.assertEquals(-1, calculateWithData(ProgrammerCalcModel(BigDecimal(-3), BigDecimal(-2), Operator.get_remainder, mode)))
            Assert.assertEquals(0, calculateWithData(ProgrammerCalcModel(BigDecimal(-2), BigDecimal(-2), Operator.get_remainder, mode)))
        }
    }
}